packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts

```diff
@@ -166,7 +165,7 @@ interface CloudFormationDeployActionProps extends CloudFormationActionProps {
    * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-iam-template.html#using-iam-capabilities
    * @default None, unless `adminPermissions` is true
    */
-  readonly capabilities?: cloudformation.CloudFormationCapabilities[];
+  readonly capabilities?: cdk.CfnCapabilities[];
```

Comment:
We should do this the usual way we change APIs. Mark it as `@deprecated` on master, and still support the old APIs. As part of the deprecated APIs work, these will be stripped off in v2.

Reply:
We can do that, but that will not achieve the desired outcome here.

If you leave this API in `@aws-cdk/aws-codepipeline-actions`, this means you need to leave the `@aws-cdk/aws-cloudformation` module inside of `aws-cdk-lib`, which means you cannot deprecate it.

So this entire PR would effectively be discarded.

Reply:
We could keep it in there, until deprecated APIs are stripped off. Once we achieve that and it's in place, we can special case the removal of this package. 

Actually, if all APIs on the `aws-cloudformation` package are deprecated, then none of its APIs will be visible in `aws-cdk-lib`, so no harm in still keeping it in.

However, it's important that all usages of the cloudformation packages be deprecated for this to work.

Reply:
OK. Let's not keep this for now then.

Reply:
Moved to #12670.